### PR TITLE
feat(leftsidebar): Show drop zones only on hover

### PR DESCRIPTION
**Pull request by shahzadaziz**
This PR allows you to set `showDropZoneOnHover` for menu items in left sidebar. 

`showDropZoneOnHover`  when true it will only show drop veil on hover. 
`showDropZoneOnHover`  when false it will show drop veil for the duration of items being dragged.

Updated: GIF to show behavior on favorites 

favoritesDragDrop2

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @shahzadaziz has signed the CLA. Thanks for the pull request!

**Comment by booc0mtaco** [comment]
more of a UX question; we show that you can't drag to other files by dimming them out. could we do the same for the left sidebar items that can't be dragged to?

File: src/features/left-sidebar/LeftSidebarDropWrapper.js [hunk 1]
```diff
@@ -9,24 +9,67 @@ type Props = {
...
+            className = '',
+            isDragging = false,
+            showDropZoneOnHover = false,
+            message = '',
+            dropTargetRef,
+            ...rest
+        } = this.props;
+        const { dropZoneHover } = this.state;
+        const shouldShowVeil = isDragging && (showDropZoneOnHover ? dropZoneHover : true);
+        const hoverEventHandlers =
```
Comment (booc0mtaco) [inline src/features/left-sidebar/LeftSidebarDropWrapper.js]: do we handle escape when dragging?

Comment (shahzadaziz) [reply src/features/left-sidebar/LeftSidebarDropWrapper.js]: We do. `isDragging` is set false.

**Comment by shahzadaziz** [comment]
> more of a UX question; we show that you can't drag to other files by dimming them out. could we do the same for the left sidebar items that can't be dragged to?

I discussed this with design. We might lean towards not showing any visual indicators at all when dragging and move towards drop zones on hover only.  Design wants it to behave similar to how some operating systems e.g Mac OS does it.

**Comment by booc0mtaco** [comment]
@shahzadaziz yeah, the on-drag-start indicators is more of an older web convention. if we aim to move away from that or not is cool.

But, if we're aiming to masquerade as having macOS-like behavior, worth nothing what the HIG recommends, RE: the question about dimming out stuff

[User Interaction: Drag and Drop](https://developer.apple.com/design/human-interface-guidelines/macos/user-interaction/drag-and-drop/)

> **Clearly indicate when dropping isn’t possible**. During a drag, users should know when they can and can’t drop content. Some apps communicate this by displaying an insertion point, highlighting, or acceptance pointer badge only when dropping is possible—in other words, the lack of visual feedback indicates when dropping isn’t possible. Other apps display an explicit not allowed badge, such as a circle with a slash.

this would dim out the things that can't be dragged to in the left side bar, or change the cursor, etc.

> **Highlight a new destination when content is dragged above it**. When there are multiple destinations, highlight one at a time. When the source and destination containers are the same, highlighting is unnecessary. For example, in Finder, moving a document icon to a different location in the same folder window doesn’t highlight the folder window because doing so would cause confusion. In some cases, highlighting may occur if content is dragged completely out of a container and then reenters it. Highlighting should be removed when content is no longer positioned above a destination.

we're doing this stuff with your PR, which is nice

**Comment by shahzadaziz** [comment]
@booc0mtaco Thanks for sharing this article. I will run the question of dimming out things that are not drop zones by design team. We can determine what kind of visual treatment we want to have for these and it should be consistent between Item list and Left sidebar. We can track this as an incremental improvement separately.

**Comment by priyajeet** [comment]
Make sure you are sending PRs from your fork and not pushing branches to main repo

**Pull request closed without merging**